src/software/ai/hl/stp/tactic/dribble/dribble_fsm.h

```diff
@@ -0,0 +1,220 @@
+            // experimentally
+            static const double BALL_CLOSE_TO_DEST_THRESHOLD = 0.1;
+            return ((event.common.world.ball().position() - getBallDestination(event))
+                        .length() < BALL_CLOSE_TO_DEST_THRESHOLD);
+        };
+
+        /**
+         * Action to update to get possession of the ball
+         *
+         * If the ball is moving quickly, then move in front of the ball
+         * If the ball is moving slowly, then chase the ball
+         *
+         * @param event DribbleFSM::Update
+         */
+        const auto get_possession = [this](auto event) {
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            Point intercept_position =
+                findInterceptionPoint(event.common.robot, event.common.world.ball(),
+                                      event.common.world.field());
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), intercept_position, face_ball_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT));
+        };
+
+        /**
+         * Action to update to get possession of the ball
+         *
+         * @param event DribbleBallFSM::Update
+         */
+        const auto dribble = [this, ball_at_dest](auto event) {
+            // Threshold to determine if the robot has the expected orientation
+            static const Angle ROBOT_ORIENTATION_CLOSE_THRESHOLD = Angle::fromDegrees(5);
+            // helper calculations
+            auto ball_position    = event.common.world.ball().position();
+            auto ball_destination = getBallDestination(event);
+            Angle to_destination_orientation =
+                (ball_destination - ball_position).orientation();
+
+            // Default destination and orientation assume ball is at the destination
+            // pivot to final face ball destination
+            Angle target_orientation = getFinalFaceBallOrientation(event);
+            Point target_destination =
+                convertBallPositionToRobotPosition(ball_destination, target_orientation);
+
+            if (!ball_at_dest(event))
+            {
+                if (to_destination_orientation.minDiff(event.common.robot.orientation()) <
+                    ROBOT_ORIENTATION_CLOSE_THRESHOLD)
+                {
+                    // dribble towards ball destination
+                    target_destination = convertBallPositionToRobotPosition(
+                        ball_destination, to_destination_orientation);
+                    target_orientation = to_destination_orientation;
+                }
+                else
+                {
+                    // pivot to face ball destination
```

Comment:
If this is else statement just pivoting to face the final orientation (the ball destination is the ball's current position), what is the difference from lines 181-183 (assumes the ball is at the destination and just needs to pivot)?

Reply:
the idea is to pivot in place with the ball, I've clarified the comment